### PR TITLE
xdebug: update to 3.0.4, remove php-pear in favour of pear (we bootstrapping now folks)

### PR DESCRIPTION
**Pull request by the-eater**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by the-eater** [comment]
As a note, I am assigning myself as maintainer of `php` and `xdebug` I've been the major force doing the bigger upgrades for those the last few years, and the original maintainers haven't touched it in 4 years and considering my open RFC it would make sense. I am not expecting anybody to have an issue with this.

File: srcpkgs/pear/template [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+_version_console_getopt=1.4.3
+_version_archive_tar=1.4.13
+_version_structures_graph=1.1.1
+_version_php_archive=0.14.0
+wrksrc="pear-core-${version}"
+hostmakedepends="php"
+depends="php"
+conf_files="/etc/php/pear.conf"
+short_desc="PEAR - PHP Extension and Application Repository"
+maintainer="eater <=@eater.me>"
```
Comment (ericonr) [inline srcpkgs/pear/template:15]: Huh, `=` is legal? :o

File: srcpkgs/pear/template [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	done
+
+	# PHP_Archive classes are expected to be at PHP/Archive but it is not
+	mv "go-pear-tarballs/src/PHP_Archive-${_version_php_archive}" go-pear-tarballs/src/PHP
+}
+
+post_patch() {
+	# PEAR would normally replace these, but we're pearless at this point
+	vsed -i 's:\(.\)@data_dir@/PHP_Archive/:__DIR__ . \1/../:;s:@API-VER@:1.1.0:' \
+		go-pear-tarballs/src/PHP/Archive/Creator.php
```
Comment (ericonr) [inline srcpkgs/pear/template:67]: A peerless substitution :D

File: srcpkgs/php/template [hunk 3]
```diff
@@ -1,15 +1,15 @@
 # Template file for 'php'
 pkgname=php
 version=7.4.14
-revision=3
+revision=4
```
Comment (ericonr) [inline srcpkgs/php/template:4]: Maybe add an install.msg for php-pear saying that Pear has been removed? And make `php-pear` a dumy transitional package for now.

We can't make it depend on `pear` due to cyclic deps, and `pear` can't have a `php-pear` subpackage without a huge list of reverts.

**Comment by ericonr** [comment]
And I think in the commit message for `pear` you should explain why it's needed as a separate package too :)

**Pull request closed without merging**